graph/src/data/query/result.rs

```diff
@@ -97,54 +172,65 @@ impl QueryResult {
             .body(T::from(json))
             .unwrap()
     }
+}
 
-    /// Combine all the data into one `q::Value`. This method might clone
-    /// all of the data in this result
-    fn take_data(self) -> Option<q::Value> {
-        fn take_or_clone(value: Arc<Data>) -> Data {
-            Arc::try_unwrap(value).unwrap_or_else(|value| value.as_ref().clone())
-        }
+/// The result of running a query, if successful.
+#[derive(Debug, Serialize)]
+pub struct QueryResult {
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialize_data"
+    )]
+    data: Option<Data>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    errors: Vec<QueryError>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialize_extensions"
+    )]
+    extensions: Option<q::Value>,
```

Comment:
We should really nuke `extensions` since it is always `None`

Reply:
Yay done